Change to LFRic Macros scripts for git

james-bruten-mo opened the pull request:
Update the lfric macros scripts for the git migration.

This branch was used to track other git migration changes which have been merged in other PRs. The only remaining changes are to the lfric macro scripts.

yaswant commented on lfric_macros/apply_macros.py:
```diff
@@ -72,31 +73,29 @@ def check_environment():
-            "[WARN] - Could not find the fcm root path for the apps working copy. "
-            "Defaulting to assuming the provided path in the root path."
+            "[WARN] - Could not find the git clone root for the apps working copy. "
+            "Defaulting to assuming the provided path is the root path."
```
```suggestion
            "[WARN] - Unable to locate the git clone root for the apps working copy."
            "Assuming the given path is the root directory by default."
```

yaswant commented on lfric_macros/apply_macros.py:
```diff
@@ -347,76 +345,58 @@ def get_dependency_paths(self, source, repo):
-        # Check that the source looks like an fcm keyword, raise an error if not
-        if "fcm:" not in source:
+        # Check that the source looks like a github URL, raise an error if not
+        if ".git" not in source:
```
Hmmm.. should we really be expecting an explicit `.git` suffix in GH URL?

yaswant commented on lfric_macros/release_lfric.py:
```diff
@@ -87,14 +88,19 @@ def set_dependency_path(args):
+            line = f"{line[0]}source:{hostname}:{os.path.abspath(args.core)}\n"
+        elif in_core and "ref:" in line:
+            line = line.split("ref:")
+            line = f"{line[0]}ref:"
```
as above

yaswant reviewed (commented):
Also a general comment about duplicating `run_command()` in many places.. It might be a good idea to have a generic class. And prefer `shlex.split(command)` over `command.split()` _in a separate pr_.

Otherwise the changes look good to me.

james-bruten-mo commented on lfric_macros/release_lfric.py:
```diff
@@ -87,14 +88,19 @@ def set_dependency_path(args):
+            in_core = True
+        elif in_core and "source:" in line:
+            line = line.split("source:")
+            line = f"{line[0]}source:{hostname}:{os.path.abspath(args.core)}\n"
```
Done

james-bruten-mo commented:
Thanks Yash, all done. I didn't know about shlex.split which is very nice.
I've also updated some git add commands so they run in the correct location which was causing an issue
